docs/DeveloperGuide.md

```diff
@@ -1,38 +1,327 @@
+
+**User Interaction Flow**
+
+The command syntax for generating gift suggestions would be expanded to allow for optional gender specification: `gift`, `gift male`, or `gift female`. This modification enables the application to cater to a broader range of user intentions and preferences, enhancing the personalised nature of the gift suggestion feature.
+
+- By enabling gender-specific gift suggestions, the application can provide more targeted and relevant options, increasing the likelihood of user satisfaction.
+- This enhancement also retains flexibility by allowing users to opt for non-gendered suggestions when preferences are not specified or when seeking more universally appealing gift ideas.
+
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+Couples in Singapore
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+In today's fast-paced world, maintaining a healthy and exciting relationship can often be challenging for couples. Balancing work, personal responsibilities, and quality time with a partner requires effort and planning. Many couples struggle to come up with new and interesting date ideas that cater to both partners' preferences, leading to a routine that can become stale and uninspiring. This is where Flirt and Fork steps in to rejuvenate the love life of couples by offering a seamless, engaging, and personalised date planning experience.
 
 ## User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|v1.0|first-time user|see a help message|know how to use the features within the app & its parameters|
+|v1.0|busy user|effortlessly generate date ideas|not waste time endlessly searching for ideas|
+|v1.0|budget conscious user|generate a date itinerary based on price|choose when to splurge and when to save|
+|v1.0|user looking for variety|view a variety of date activities|enjoy diverse date experiences|
+|v1.0|experienced user of the platform|save some of my preferred activities|reference back to them in future|
 
 ## Non-Functional Requirements
 
 {Give non-functional requirements}
 
 ## Glossary
 
-* *glossary item* - Definition
+
+Generate Idea Feature: A feature within the application that generates random date ideas consisting of food and activity suggestions.
+
+Smart Itinerary Generation Feature: A proposed feature that creates personalized itineraries based on user preferences, ensuring variety and suitability.
+
+History Tracking Feature: A proposed feature that keeps a record of past date activities, allowing users to track their date history and preferences.
```

Comment:
As mentioned above, I don't think it tracks each date history, it seems to just track which activities are completed?